src/typing/flow_js.ml

```diff
@@ -6647,6 +6647,15 @@ struct
         | ( DefT (reason, _, ArrT (ArrayAT (t, _))),
             (GetPropT _ | SetPropT _ | MethodT _ | LookupT _) ) ->
           rec_flow cx trace (get_builtin_typeapp cx ~trace reason "Array" [t], u)
+        (*************************)
+        (* Tuple "length" access *)
+        (*************************)
+        | ( DefT (reason, trust, ArrT (TupleAT (_, ts))),
+            ( LookupT (_, _, _, Named (_, "length"), ReadProp { use_op = _; obj_t = _; tout })
```

Comment:
Is the `LookupT` match necessary? It's typically used for traversing prototype chains. (None of the tests are affected by removing it.)

Reply:
Yep, I was also not sure if it needs to stay.  You probably have a better understand on how it works.
There was a discussion in `#contributor-support` and @goodmind suggested we keep it.
@jbrown215 also commented, but I am not sure the conclusion was very solid.

I can easily remove it if it does not matter.  And if it does, it would be good to add a test.  But, unfortunately, I do not know what the code should look like for `LookupT` to be triggered and for the `GetPropT` not to.

Here is a discussion:
```
ilya-bobyr09/16/2019
Actually, if I replace LookupT with GetPropT, the tests pass %)
would you recommend a function, module, and/or a comment to look at to better understand how all these types are used?

goodmind09/16/2019
no idea, but I would keep LookupT because it's more general than GetPropT or ask when someone from flow team is online
there isn't much description of differences between them

[...]

jbrown21509/16/2019
@goodmind: I'm 99% certain that GetPropT eventually delegates to LookupT
but LookupT assumes certain properties about the left-hand-side of the flow, so GetPropT first does some work on that

goodmind09/16/2019
@jbrown215 on tuples it is just handle-all case
so idk how LookupT, GetPropT works there
I thought LookupT is only for prototype chain
And tuples aren't implemented like this

jbrown21509/16/2019
LookupT definitely traverses the prototype chain. I'm pretty unfamiliar with the tuple implementation

goodmind09/16/2019
that's all
so it looks like you have to duplicate matchers for LookupT and GetPropT if you want to overwrite some property, objects are more complex than this
```

Any idea what can be a candidate for a test for `LookupT`?

Reply:
Okay, I took a closer look, and I think we should lose the `LookupT` check. The only case where I managed to see a difference is with the `--no-flowlib` flag. Take the following code for example
```
declare var a: [string];
(a.length === 1) ? true : false;

declare var b: Array<string>;
(b.length === 1) ? true : false;
```
With your changes, I get 
```
$ flow check tuples/ --no-flowlib
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ tuples/file.js:7:2

Property length is missing in array type [1].

      4│ (a.length === 1) ? true : false;
      5│
 [1]  6│ declare var b: Array<string>;
      7│ (b.length === 1) ? true : false;
      8│

Found 1 error
```
In line 4, the `ArrT (TupleAT _), LookupT ("length")` rule fires and prevents the error even though we are not importing the `Array` defs from `lib/core.js`. For consistency with what we do for arrays, we probably should error in this case as well.

Without the flow `--no-flowlib` and without the `LookupT` case, the `length` property gets looked up in the `Array` prototype, so we don't report any errors.

Reply:
Thanks for looking into it.  Removed `LookupT`.